tests/test_snipping.py

```diff
@@ -487,3 +487,141 @@ def test_cooler_snipper_with_view(request):
+
+    stack = cooltools.api.snipping.pileup(
+        clr, windows, view_df=view_df, expected_df=exp, flank=None
+    )
+    stack_pooled2 = cooltools.api.snipping.pileup(
+        clr, windows, view_df=view_df, expected_df=exp, flank=None, nproc=2
+    )
+    stack_pooled3 = cooltools.api.snipping.pileup(
+        clr, windows, view_df=view_df, expected_df=exp, flank=None, nproc=3
+    )
+    # check if they output the same result
+    assert np.array_equal(stack, stack_pooled2, equal_nan=True)
+    assert np.array_equal(stack, stack_pooled3, equal_nan=True)
+
+    # II.
+    # Example off-diagonal features, two features from annotated genomic regions:
+    windows = pd.DataFrame(
+        {
+            "chrom1": ["chr1", "chr1"],
+            "start1": [102_000_000, 107_000_000],
+            "end1": [107_000_000, 112_000_000],
+            "chrom2": ["chr1", "chr1"],
+            "start2": [107_000_000, 113_000_000],
+            "end2": [112_000_000, 118_000_000],
+        }
+    )
+    stack = cooltools.api.snipping.pileup(
+        clr, windows, view_df=view_df, expected_df=exp, flank=None
+    )
+    stack_pooled2 = cooltools.api.snipping.pileup(
+        clr, windows, view_df=view_df, expected_df=exp, flank=None, nproc=2
+    )
+    stack_pooled3 = cooltools.api.snipping.pileup(
+        clr, windows, view_df=view_df, expected_df=exp, flank=None, nproc=3
+    )
+    # check if they output the same result
+    assert np.array_equal(stack, stack_pooled2, equal_nan=True)
+    assert np.array_equal(stack, stack_pooled3, equal_nan=True)
+
+    # III.
+    # Example off-diagonal features, one region outside the view:
+    windows = pd.DataFrame(
+        {
+            "chrom1": ["chr1", "chr1"],
+            "start1": [90_000_000, 105_000_000],
+            "end1": [95_000_000, 110_000_000],
+            "chrom2": ["chr1", "chr1"],
+            "start2": [105_000_000, 110_000_000],
+            "end2": [110_000_000, 115_000_000],
+        }
+    )
+    stack = cooltools.api.snipping.pileup(
+        clr, windows, view_df=view_df, expected_df=exp, flank=None
+    )
+    stack_pooled2 = cooltools.api.snipping.pileup(
+        clr, windows, view_df=view_df, expected_df=exp, flank=None, nproc=2
+    )
+    stack_pooled3 = cooltools.api.snipping.pileup(
+        clr, windows, view_df=view_df, expected_df=exp, flank=None, nproc=3
+    )
+    # check if they output the same result
+    assert np.array_equal(stack, stack_pooled2, equal_nan=True)
+    assert np.array_equal(stack, stack_pooled3, equal_nan=True)
+
+
+    # IV.
+    # Example on-diagonal features, not valid bedframes (start>end):
+    windows = pd.DataFrame(
+        {
+            "chrom": ["chr1", "chr1"],
+            "start": [107_000_000, 108_000_000],
+            "end": [102_000_000, 113_000_000],
+        }
+    )
+    with pytest.raises(ValueError):
+        stack = cooltools.api.snipping.pileup(clr, windows, view_df, exp, flank=None)
```

Comment:
probably only need one of these 3 tests-- also they are under the same with statement